### ISSUE TITLE
Panic on startup: 'gl function was not loaded'

### ISSUE THREAD

**Issue by zacps** [zacps/alacritty#12]
From [jwilm/alacritty#28](https://github.com/jwilm/alacritty/issues/28#issuecomment-369619638).

Now however I am getting a different error when I try to run alactritty.exe.

```rust
Welcome to Alacritty.                                                                                                                                            
Configuration loaded from C:\Users\gavin\alacritty.yml                                                                                                           
device_pixel_ratio: 1                                                                                                                                            
width: 1024, height: 768                                                                                                                                         
thread 'main' panicked at 'gl function was not loaded', E:\Development\alacritty\target\release\build\alacritty-aabf91f3f7e36563\out/gl_bindings.rs:20617:13     
stack backtrace:                                                                                                                                                 
   0: std::panicking::default_hook::{{closure}}                                                                                                                  
             at C:\projects\rust\src\libstd\panicking.rs:386                                                                                                     
   1: std::panicking::rust_panic_with_hook                                                                                                                       
             at C:\projects\rust\src\libstd\panicking.rs:563                                                                                                     
   2: std::panicking::begin_panic<str*>                                                                                                                          
             at C:\projects\rust\src\libstd\panicking.rs:538                                                                                                     
   3: alacritty::gl::missing_fn_panic                                                                                                                            
             at .\build\alacritty-aabf91f3f7e36563\out\gl_bindings.rs:20617                                                                                      
   4: alacritty::main                                                                                                                                            
             at E:\Development\alacritty\src\main.rs:45                                                                                                          
   5: std::rt::lang_start::{{closure}}<()>                                                                                                                       
             at C:\projects\rust\src\libstd\rt.rs:74                                                                                                             
   6: main                                                                                                                                                       
   7: __scrt_common_main_seh                                                                                                                                     
             at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283                                                                                       
   8: BaseThreadInitThunk    ```

**Comment by zacps** [comment]
@gavinhow Could you post your GPU, driver version and opengl information (using something like [opengl extensions viewer](http://realtech-vr.com/admin/glview))?

**Comment by gavinhow** [comment]
GPU: Radeon RX480 8GB
Driver Version: Adrenalin 18.2.1

Below is some information I managed to pull from the tool you linked. I wasn't exactly sure what you were looking for so just took what I thought looked useful.
````
**Open GL**
Verson: 4.5
Driver version: 
    23.20.150173010
    Catalyst 17.5.17.03
    31-Jan-18
````
````
**GL Report**
Renderer: Radeon (TM) RX 480 Graphics
Vendor: ATI Technologies Inc.
Memory: 4095 MB
Version: 4.5.13506 Core Profile Forward-Compatible Context 23.20.15017.3010
Shading language version: 4.50


Max texture size: 16384 x 16384
Max vertex texture image units: 32
Max texture image units: 32
Max geometry texture units: 32
Max anisotropic filtering value: 16
Max viewport size: 16384 x 16384
Max Clip Distances: 8
Max samples: 8
````

**Comment by zacps** [comment]
This is weird, I had a similar issue during development because the initialisation code was defining the load method before making the opengl context current which doesn't work on windows. I'm not sure what else could the issue could be.
Could you try running the examples from [gl-rs](https://github.com/brendanzab/gl-rs)?

**Comment by prabirshrestha** [comment]
I'm having the same issue. Full graphics info can be found at https://pastebin.com/xVbKfm8S

**Comment by prabirshrestha** [comment]
update git repo and rust stable `rustc 1.24.1 (d3ae9a9e0 2018-02-27)` and ran `cargo update` and `cargo build --release`. I no longer this error but now I see the window with white screen open and close. @zacps can you share the `alacritty.yml` and the exe file.

**Comment by zacps** [comment]
I have been doing all my builds on nightly, that might make a difference.
Regardless here's a debug build of the latest commit and my working alacritty.yml file.

---
[alacritty_build_64434a.zip](https://github.com/zacps/alacritty/files/1774171/alacritty_build_64434a.zip)

**Comment by hrach** [comment]
The build crashes quite fast, I screenshoted the error which says: "Error creating GL context; Couldn't find any pixel format that matches the criterias."

**Comment by prabirshrestha** [comment]
@zacps your build works. Mine didn't work from rust stable. Will try rust nightly and see if that works.
Also seems like good idea to fix this. https://github.com/zacps/alacritty/issues/1. I had issues with my config.

**Comment by zacps** [comment]
@hrach I found a bit of information on this issue in https://github.com/PistonDevelopers/piston/issues/1202 and [this gist](https://gist.github.com/azriel91/7a33da3061fa1fb734083e8d22451f91).
First try the latest commit (I updated glutin from a *very* old version) and if that doesn't work try disabling vsync in `src/window.rs:188`.

**Comment by prabirshrestha** [comment]
@zacps Lastest master seems to work. I didn't need to update src/windows.rs

**Comment by zacps** [comment]
Excellent, hopefully the same works for @gavinhow and hrach.

**Comment by gavinhow** [comment]
Seems to be working on the latest commit which is great.

I had to add `winpty-agent.exe` which I got from your debug build above to the target folder as otherwise it just flashed up and closed straight away.

Good that it is now working though thanks for your help. I will be following this closely and trying out new builds as they come!

edit: Might also be worth noting that I got it working on rust stable.
